apps/labrinth/src/database/models/user_item.rs

```diff
@@ -599,13 +599,25 @@ impl DBUser {
             sqlx::query!(
                 "
                 DELETE FROM reports
-                WHERE user_id = $1 OR reporter = $1
+                WHERE reporter = $1
                 ",
                 id as DBUserId,
             )
             .execute(&mut **transaction)
             .await?;
 
+            sqlx::query!(
+                "
+                UPDATE reports
+                SET user_id = $1
+                WHERE reporter = $2
+                ",
+                deleted_user as DBUserId,
+                id as DBUserId,
+            )
+            .execute(&mut **transaction)
+            .await?;
```

Comment:
Wait, actually this looks a bit strange: here you're changing the reports reported by the to-be-deleted user to be reported by the ghost user, but a few lines above you delete all reports where the reporter was the to-be-deleted user, so this query will never update any rows. What was your intention here?

Reply:
That makes sense 👍